src/Tempest/CommandBus/src/AsyncCommandRepositories/FileCommandRepository.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\CommandBus\AsyncCommandRepositories;
+
+use Tempest\CommandBus\CommandRepository;
+use Tempest\CommandBus\Exceptions\CouldNotResolveCommand;
+use function Tempest\Support\arr;
+
+final readonly class FileCommandRepository implements CommandRepository
+{
+    public function store(string $uuid, object $command): void
+    {
+        $payload = serialize($command);
+
+        file_put_contents(__DIR__ . "/../stored-commands/{$uuid}.pending.txt", $payload);
+    }
+
+    public function findPendingCommand(string $uuid): object
+    {
+        $path = __DIR__ . "/../stored-commands/{$uuid}.pending.txt";
+
+        if (! file_exists($path)) {
+            throw new CouldNotResolveCommand($uuid);
+        }
+
+        $payload = file_get_contents($path);
+
+        return unserialize($payload);
+    }
+
+    public function markAsDone(string $uuid): void
+    {
+        $path = __DIR__ . "/../stored-commands/{$uuid}.pending.txt";
+
+        unlink($path);
+    }
+
+    public function markAsFailed(string $uuid): void
+    {
+        rename(
+            from: __DIR__ . "/../stored-commands/{$uuid}.pending.txt",
+            to: __DIR__ . "/../stored-commands/{$uuid}.failed.txt",
+        );
+    }
+
+    public function getPendingCommands(): array
```

Comment:
We'll run into performance issue if we're always returning _all_ pending commands. I think it outside the scope of this PR to optimize that though. Let's first play a bit with it as-is and see what other problems surface